### DISCUSSION TITLE
Can't deploy K3s other than /var/lib/rancher

### DISCUSSION THREAD

**Discussion by busyboy77** [k3s-io/k3s#7412] [Q&A]
**Environmental Info:**
K3s Version: 

k3s version v1.25.4+k3s1 (0dc63334)
go version go1.19.3

Node(s) CPU architecture, OS, and Version: 

Cluster Configuration:
Linux devops55.ef.com 4.18.0-305.25.1.el8_4.x86_64 #1 SMP Mon Oct 18 14:34:11 EDT 2021 x86_64 x86_64 x86_64 GNU/Linux

**Describe the bug:**
when deployed using --data-dir option pointing to a location other than /var/lib/rancher, it still creates all directories' structure in default locations.

consider a systems environment where the 

* customer is not agreed to provision space on /var ( with /var only 20GiB)
* Or the Infra environment doesn't allow to use /var for K3s deployments and you are just allocated space on a different mount point.

Now, what happens after you deploy the --data-dir changed, it still deploys the solution to default locations under  /run/k3s, /var/lib/kubelet/pods, /var/lib/rancher and the solution hangs due to space constraints .

the guide to use containerd configuration file (config.toml.tmpl) is too complex for relocation of the pods to some other location. 

**Steps To Reproduce:**

a complete relocation path should be there to keep the data in any other location than the default /var 

- Installed K3s: 

**Expected behavior:**

**Actual behavior:**

**Additional context / logs:**

**Comment by brandond** [comment]
> when deployed using --data-dir option pointing to a location other than /var/lib/rancher, it still creates all directories' structure in default locations.

Are you sure you're setting the option correctly? I am unable to reproduce the behavior you're describing.

**Reply by brandond** [reply]
@busyboy77 I would suggest you try configuring the data-dir via config.yaml so that you don't have to pass it on the CLI every time you run a k3s command. I am not aware of any other issues that would cause the data-dir to not be respected.

**Comment by davinkevin** [comment]
I've faced the same issue, with the cluster not able to start properly. 

I was in a situation with the cluster running in `/var` and I decided to change it  with the extra parameter (using `--data-dir`) + `mv` of all files to the new location. 

In my context, the server never started, was stuck on `k3s starting` and `syslog` was full of `Apr 28 05:08:20 localhost k3s[3719]: time="2023-04-28T05:08:20+02:00" level=info msg="Waiting for API server to become available"` until the main disk was full (100Gb of logs, 1 every second). 

And like described, some files were still in `/var/lib/rancher/k3s`.

**Reply by brandond** [reply]
Moving things after the fact is not quite the same as starting up with the --data-dir set properly. We do not test or support moving the data-dir after the fact, although I do suspect that it should work if you work at it a bit.
